media-proxy/src/api_server_tcp.cc

```diff
@@ -73,38 +71,32 @@ void* msg_loop(void* ptr)
             /* read parameters */
             buffer = (char*)malloc(msg.command.data_len);
             if (buffer == NULL) {
-                INFO("Out of Memory.");
+                ERROR("(char*)malloc(msg.command.data_len) in msg_loop() failed. Out of Memory.");
                 continue;
             } else {
+                ret = 0;
                 int bytesRead = 0;
-                while (bytesRead < msg.command.data_len) {
-                    int ret = read(conn->sock, buffer + bytesRead, msg.command.data_len - bytesRead);
-                    if (ret <= 0) {
-                        INFO("Failed to read command parameters.");
-                        free(buffer);
-                        buffer = NULL; // Reset the pointer to indicate failure
-                        break; // Exit the loop on failure
-                    }
-                    bytesRead += ret;
+                while (bytesRead < msg.command.data_len && ret >= 0) {
+                    ret = read(conn->sock, buffer + bytesRead, msg.command.data_len - bytesRead);
```

Comment:
I suppose, if you're still using a blocking socket, there is a high probability that this `read` call will be blocked when a termination signal occurs. And you'll never get a graceful shutdown in this case.

Reply:
You are right here - I had similar thoughts. But on the other hand, in this loop I have included `keepRunning` from the main loop and `sessionKeepRunning` so I am/was looking for a way to force this thread to be shut down in case the client have disconnected on the other end of socket and cleanup after everything but at the moment I am looking for your//other reviewer possible ideas.